### PR TITLE
Update docker build

### PR DESCRIPTION
**Pull request by StephenOman**
This PR updates the Docker Distribution capability

- Removes old Ubuntu distros
- Add Ubuntu 22.04 (Jammy)
- Update build for multi-architecture deployment (x86_64, amd64 and arm64)
- Changed README instructions

### CODE REVIEW

File: docker/Dockerfile-focal [hunk 1]
```diff
@@ -19,11 +33,16 @@ RUN apt-get update && apt-get install -yq \
 
 WORKDIR /opt/conda_setup
 
-RUN curl -o miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
-     chmod +x miniconda.sh && \
-     ./miniconda.sh -b -p /opt/conda && \
-     /opt/conda/bin/conda install -y python=$PYTHON_VERSION && \
-     /opt/conda/bin/conda clean -ya
+RUN case ${TARGETPLATFORM} in \
```
Comment (heiner) [inline docker/Dockerfile-focal:36]: These days, I'd probably use uv and nothing else but we don't need to change this in this PR.

**Review by heiner — APPROVED** [review]
Nice, thanks! 

Remind me what I need to do re: automatic building of these images?

File: docker/Dockerfile-focal (hunk 1 above)
Comment (heiner) [reply docker/Dockerfile-focal:36]: That would probably help save GitHub minutes too as the non-nethack parts are likely the majority of the build time.

Comment (StephenOman) [reply docker/Dockerfile-focal:36]: What's uv?

**Comment by StephenOman** [comment]
> Remind me what I need to do re: automatic building of these images?

Depends on which solution we want to pursue, Docker Hub or GitHub Container Registry. For now I suggest letting users build their own Docker images per instructions and look to automating the image building to coincide with NLE-1.0.0.

File: docker/Dockerfile-focal (hunk 1 above)
Comment (heiner) [reply docker/Dockerfile-focal:36]: https://astral.sh/blog/uv

I'm already using it in one of the tests, [here](https://github.com/heiner/nle/blob/main/.github/workflows/lint_python.yml#L18).

It's from the `ruff` developer and one of the many Rust-for-Python tools that massively speed up things these days.

Comment (StephenOman) [reply docker/Dockerfile-focal:36]: On my M3 Mac, the Docker build takes 15 minutes and the image size is a chunky 23GB.

Comment (heiner) [reply docker/Dockerfile-focal:36]: Is it faster the second time and we just have to cache the Docker layers?

Alternatively, I think the base image with CUDA is quite heavy.

Comment (StephenOman) [reply docker/Dockerfile-focal:36]: It is faster, but still takes nearly 10 minutes, most of the time is in `RUN pip install '.[all]'` on amd64 (~400s). Rather strangely, the same step for arm64 only takes about 40s. I wonder is this due to me running the build on a Mac.

Comment (heiner) [reply docker/Dockerfile-focal:36]: Yes, that's likely. Cross compilation is rather slow on arm MacBooks.

**Pull request merged**